pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -805,3 +852,16 @@ def _customize_output_object(transformer: type[JsonSchemaTransformer], o: Output
         json_schema=json_schema,
         strict=schema_transformer.is_strict_compatible if o.strict is None else o.strict,
     )
+
+
+def _get_final_result_event(e: ModelResponseStreamEvent, params: ModelRequestParameters) -> FinalResultEvent | None:
```

Comment:
`model_request_parameters` was added to all the model `StreamingResponse` classes as it's needed here